src/lifetime_free.rs

```diff
@@ -83,13 +83,15 @@ unsafe impl<T: LifetimeFree> LifetimeFree for core::num::Wrapping<T> {}
 unsafe impl<T: LifetimeFree> LifetimeFree for core::cell::Cell<T> {}
 unsafe impl<T: LifetimeFree> LifetimeFree for core::cell::RefCell<T> {}
 
-#[cfg(feature = "std")]
-mod std_impls {
+#[cfg(feature = "alloc")]
+mod alloc_impls {
     use super::LifetimeFree;
 
-    unsafe impl LifetimeFree for String {}
+    unsafe impl LifetimeFree for alloc::string::String {}
 
-    unsafe impl<T: LifetimeFree> LifetimeFree for Box<T> {}
-    unsafe impl<T: LifetimeFree> LifetimeFree for Vec<T> {}
-    unsafe impl<T: LifetimeFree> LifetimeFree for std::sync::Arc<T> {}
+    unsafe impl<T: LifetimeFree> LifetimeFree for alloc::boxed::Box<T> {}
+    unsafe impl<T: LifetimeFree> LifetimeFree for alloc::vec::Vec<T> {}
+
+    #[cfg(target_has_atomic = "ptr")]
+    unsafe impl<T: LifetimeFree> LifetimeFree for alloc::sync::Arc<T> {}
```

Comment:
`target_has_atomic` requires Rust 1.60, which is currently higher than the MSRV of this crate. There are a few options on how to address this:

1. We could gate the impl on `Arc` behind `#[rustversion::since(1.60)]`. This could be very surprising to users since `Arc` is almost always available.
2. We could add a separate crate feature for this impl that is disabled by default. That would be pretty tedious though to use I think.
3. Make the impl unconditional and simply allow compilation to fail on targets without atomic pointers. Also unfortunate.

Those are the options I can think of anyway. Number 3 is what we already do today, so probably makes sense to stick with that. We could also potentially combine number 1 and 3 by implementing it conditionally for Rust 1.60+, and unconditionally otherwise. E.g.

```suggestion
    #[rustversion::attr(since(1.60), cfg(target_has_atomic = "ptr"))]
    unsafe impl<T: LifetimeFree> LifetimeFree for alloc::sync::Arc<T> {}
```